logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/asset/UploadPrivateAssetUseCase.kt

```diff
@@ -0,0 +1,35 @@
+package com.wire.kalium.logic.feature.asset
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.asset.AssetRepository
+import com.wire.kalium.logic.data.asset.AssetType
+import com.wire.kalium.logic.data.asset.UploadedAssetId
+import com.wire.kalium.logic.data.user.UserRepository
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.suspending
+
+interface UploadPrivateAssetUseCase {
+    /**
+     * Function allowing the upload of a user profile picture (avatar)
+     * This first will upload the data as an asset and then will link this asset with the [User]
+     *
+     * @param mimeType mime type of the asset to be sent
+     * @param assetEncryptedData binary data of the actual asset to be sent
+     * @return an [Either] tuple containing [CoreFailure] in case of an error or the [UploadedAssetId] containing the identifier of the
+     * successfully uploaded asset
+     */
+    suspend operator fun invoke(mimeType: AssetType, assetEncryptedData: ByteArray): Either<CoreFailure, UploadedAssetId>
```

Comment:
`assetEncryptedData` - that would mean we let the UI deal with encryption which I don't think we should do.

Reply:
This UseCase should definitely not be publicly visible. Check the comment above. However I imagine the asset raw data should be still passed around as a ByteArray right?

Reply:
removed it